src/machine/machine_stm32wle5.go

```diff
@@ -221,6 +225,61 @@ func (p Pin) registerInterrupt() interrupt.Interrupt {
 	return interrupt.Interrupt{}
 }
 
+// -- SPI ----------------------------------------------------------------------
+
+type SPI struct {
+	Bus             *stm32.SPI_Type
+	AltFuncSelector uint8
+}
+
+func (spi SPI) config8Bits() {
+	// Set rx threshold to 8-bits, so RXNE flag is set for 1 byte
+	// (common STM32 SPI implementation does 8-bit transfers only)
+	spi.Bus.CR2.SetBits(stm32.SPI_CR2_FRXTH)
+}
+
+func (spi SPI) configurePins(config SPIConfig) {
+	config.SCK.ConfigureAltFunc(PinConfig{Mode: PinModeSPICLK}, spi.AltFuncSelector)
+	config.SDO.ConfigureAltFunc(PinConfig{Mode: PinModeSPISDO}, spi.AltFuncSelector)
+	config.SDI.ConfigureAltFunc(PinConfig{Mode: PinModeSPISDI}, spi.AltFuncSelector)
+}
+
+func (spi SPI) getBaudRate(config SPIConfig) uint32 {
+	var clock uint32
+	switch spi.Bus {
+	case stm32.SPI1:
+		clock = CPUFrequency()
+	case stm32.SPI2, stm32.SPI3:
+		clock = CPUFrequency()
+	}
```

Comment:
I don't understand this part. In all cases the clock equals the CPU frequency? Why not write `clock := CPUFrequency()` instead?

Reply:
IIRC, this code was (quickly) borrowed from STM32F405

I guess the lines are the same because the both SPI1 and SPI2/SPI3 's bus use a 1:1 Prescaler. 
(in the current STM32WL setup) 

In the future, we may improve / tune the clocks, and the lines may turn into :

```
	switch spi.Bus {
	case stm32.SPI1:
		clock = CPUFrequency() /2 
	case stm32.SPI2, stm32.SPI3:
		clock = CPUFrequency() /4
	}
```
So we have two solutions: 

  - Add comments to explain the PRESCALERs are 1:1 in the current configuration
  - Review and complete the work on SYSCLK/BUS/SPI clocks , prescalers... but that will take time
 
As I plan to work on Power Saving for this chip in a near future, 
i'd prefer not spending too much time on the clocks right now, and juste add
some comment. 

Tell me what you feel the most confortable with . 

Olivier

Reply:
Add comment about keeping separate to support future improvements, and IMO its all good.

Reply:
I added the comment. 
Thank you all for the time spend on this PR !